xmtp_mls/src/groups/welcomes/xmtp_welcome.rs

```diff
@@ -0,0 +1,453 @@
+//! XMTP Welcome Processing
+//! Processes a new welcome from the network
```

Comment:
I Split all welcome processing into its own file & builder, with only one public method. 

The 'Process' method contains the general steps we take when processing a welcome, with the intention of having the logic be easier to follow. It seems like since welcome processing is so critical and has been a source of bugs in the past, it deserves greater attention to detail